scripts/entity/components/movement_components/player_movement_component.gd

```diff
@@ -4,58 +4,96 @@ class_name PlayerMovementComponent extends BaseMovementComponent
 ## The max horizontal speed the entity can move.
 @export var MAX_SPEED: int = 200
 ## The jump velocity of the entity.
-@export var JUMP_VELOCITY: int = -300
+@export var JUMP_VELOCITY: int = -350
 ## The horizontal movement acceleration for the entity.
 @export var ACCELERATION: int = 1500
 ## The time it takes to fully decelerate from horizontal input in seconds.
 @export_range(0, 10, 0.1, "or_greater") var DECELERATION_TIME: float = 0.2
 
+## Moth Jump variables
+@export var MOTH_JUMP_HANGTIME: float = 1.0
+@export var MOTH_JUMP_DIP_AMOUNT: float = -18.0
+@export var MOTH_JUMP_FLUTTER_DURATION: float = 0.35
+
 ## The current input direction.
 var _current_input_direction: float = 0
 ## How fast the character decelerates after loss of input.
 var _horizontal_deceleration: float = MAX_SPEED / DECELERATION_TIME
 
-# Called when the node enters the scene tree for the first time.
+## Moth Jump state variables
+var _is_moth_jumping: bool = false
+var _is_fluttering: bool = false
+var _moth_jump_timer: float = 0.0
+var _moth_jump_flutter_timer: float = 0.0
+var _moth_jump_initial_height: float = 0.0
+var _is_jump_button_held: bool = false
+
+
 func _ready() -> void:
-	#do nothing
 	pass
 
-
 ## Applies movement to the character.
 func _physics_process(delta: float) -> void:
 	super._physics_process(delta)
-	apply_gravity(delta)	
+	apply_gravity(delta)
 	apply_horizontal_movement(delta)
 	move_and_slide()
 
-
 func on_movement_input(horizontal_vector2: float) -> void:
 	_current_input_direction = horizontal_vector2
 
-
 func on_jump_input_started() -> void:
 	if is_on_floor():
 		velocity.y = JUMP_VELOCITY
 		on_jump_started.emit()
-
+	elif not _is_moth_jumping:
+		# Start Moth Jump
+		_is_moth_jumping = true
+		_is_fluttering = true
+		_moth_jump_timer = 0.0
+		_moth_jump_flutter_timer = 0.0
+		_moth_jump_initial_height = global_position.y
+		_is_jump_button_held = true
 
 func on_jump_input() -> void:
-	# do nothing
-	pass
-
+	_is_jump_button_held = true
 
 func on_jump_input_cancelled() -> void:
-	# do nothing
-	pass
-
+	_is_jump_button_held = false
 
 ## Moves the entity based on the current input direction.
 func apply_horizontal_movement(delta: float) -> void:
- 	# Direction button active
-	if _current_input_direction:
+	if _current_input_direction != 0:
```

Comment:
Why the change? Personally felt the original was more readable.

Reply:
I personally prefer this since it is more explicit in what it is doing and leaves less room for interpretation. I think either way is fine though.

Reply:
Yeah for verbosity, I don't really care though, whichever works